spreadsheets/spread-sheet.ts

```diff
@@ -0,0 +1,77 @@
+import Sheets from 'node-sheets';
+
+export class SpreadSheet {
+  public static sheetTabularData = {};
+  private static lastUpdate = null;
+  private static currUpdateTime = null;
+  private static googleSheet: Sheets;
+  private static sheetAuthData: any;
+
+  private constructor() {}
+
+  /**
+   * Initialize the spreadsheet
+   * @param sheetId
+   * @param authData
+   */
+  public static initSpreadSheet(sheetId: string, authData: any) {
+    SpreadSheet.googleSheet = new Sheets(sheetId);
+    SpreadSheet.sheetAuthData = authData;
+  }
+
+  /**
+   * Check for updates in the spreadsheet. Returns a Promise that resolves to a boolean.
+   */
+  public static checkForUpdate = async (): Promise<boolean> => {
```

Comment:
I think we should change the responsibilities here a little
SpreadSheet class should expose a function to fetch the data (like it does in `fetchUpdatesFromGoogleSheet` perhaps rename to `fetchDataFromGoogleSheet` because it's not about _updates_). It should also expose a method to fetch last update time from Google. It should also cache the results of `fetchDataFromGoogleSheet`.

`spreadsheet-updater` then should store the last time data was fetched and also call fetchDataFromGoogleSheet in case data has changed.

So `spreadsheet` is allowing the functionality to fetch data and also get the last time the spreadsheet was updated
`spreadsheet-updater` is responsible of knowing when was the last time **we** updated the spreadsheet and also in charge of timing the checks to see when the last time the Google sheet was really updated

Reply:
So we basically don't need  `checkForUpdate` function in here

Reply:
Made some changes.

1. Added **getLastUpdateTime** in ```spreadsheet```.
2. Added *isSpreadSheetUpdated*** in ```spreadsheet-updater```.